Restructure and clarify interactive docs (#2981)
vega/altair: doc/user_guide/interactions.rst

>     chart.encode(x='Acceleration:Q') | chart.encode(x='Miles_per_Gallon:Q')

Author: I agree with this and updated the examples on the page

        brush
    )

    chart | chart.encode(x='Acceleration:Q')

Filtering Data
^^^^^^^^^^^^^^

Using a selection parameter to filter data works in much the same way
as using it within ``condition``,
For example, in ``transform_filter(brush)``,
we are again using the selection parameter ``brush`` as a predicate.
Data points which evaluate to ``True`` (i.e., data points which lie within the selection) are kept,
and data points which evaluate to ``False`` are filtered out.

It is not possible to both select and filter in the same chart,
so typically this functionality will be used when at least two sub-charts are present.
In the following example,
we attach the selection parameter to the upper chart,

> and then filter data in the lower chart based selection in the upper chart.

Reviewer: ```suggestion
and then filter data in the lower chart based on the selection in the upper chart.
```

and try typing one into the search box, e.g. ``vw pickup``
to see the point highlighted.

.. altair-plot::

    search_input = alt.selection_point(
        value='',
        fields=['Name'],
        bind=alt.binding(
            input='search',
            placeholder="Car model",
            name='Search ',
        )
    )
    alt.Chart(data.cars.url).mark_point(size=60).encode(
        x='Horsepower:Q',
        y='Miles_per_Gallon:Q',
        tooltip='Name:N',
        opacity=alt.condition(
            search_input,
            alt.value(0.8),
            alt.value(0.1)
        )
    ).add_params(
        search_input
    )

It is not always useful to require an exact match to the search syntax,
and when we will be learning about :ref:`expressions`,
we will see how we can match partial strings via a regex instead.

Data-Driven Comparisons
-----------------------

So far we have seen the use of selections
to lookup points with precisely matching values in our data.
This is often useful,
but sometimes we might want to make a more complex comparison
than an exact match.
For example,
we might want to create a condition
where we want to access the value of a slider

> to use directly in a comparison with values of the data.

Reviewer: ```
For example,
we might want to create a condition
where we want to access the value of a slider
to use directly in a comparison with values of the data.
```
It is a bit hard for me to understand this sentence, especially the last part. After reading it two times I get what you mean. Maybe split in two sentences?

.. altair-plot::

    dropdown = alt.binding_select(
        options=['Miles_per_Gallon', 'Displacement', 'Weight_in_lbs', 'Acceleration'],
        name='X-axis column '
    )
    xcol_param = alt.param(
        value='Miles_per_Gallon',
        bind=dropdown
    )

    alt.Chart(data.cars.url).mark_circle().encode(
        x=alt.X('x:Q', title=''),
        y='Horsepower:Q',
        color='Origin:N'
    ).transform_calculate(
        x=f'datum[{xcol_param.name}]'
    ).add_params(
        xcol_param
    )

> It was possible to achieve something similar before the introduction of parameters in Altair 5, but the spec for this is more involved so the solution above is to prefer:
> 
> .. altair-plot::
> 
>     # Similar workaround for earlier versions of Altair
>     source = data.cars().melt(id_vars=['Origin', 'Name', 'Year', 'Horsepower', 'Cylinders'])
>     dropdown_options = source['variable'].drop_duplicates().tolist()
> 
>     dropdown = alt.binding_select(
>         options=dropdown_options,
>         name='X-axis column'
>     )
>     selection = alt.selection_point(
>         fields=['variable'],
>         value=[{'variable': dropdown_options[0]}],
>        # init={'variable': dropdown_options[0]},  # For Altair 4
>         bind=dropdown
>     )
> 
>     alt.Chart(source).mark_circle().encode(
>         x=alt.X('value:Q', title=''),
>         y='Horsepower',
>         color='Origin',
>     ).add_params(
>         selection
>     ).transform_filter(
>         selection
>     )

Reviewer: ```suggestion
```
I think we can delete this part. Just stick to the new routine of altair v5. This page is so extensive already.

for simple interactions like this one
since they can also be accessed in expression strings
as we saw above.
Similarly,
it is often possible to use equality statements
such as ``alt.datum.xval == selector`` to lookup exact values
but it is often more convenient to switch to a selection parameter
and specifying a field/encoding.

Logic-Driven Comparisons
------------------------

> Data-driven lookups and comparisons are very helpful
> but sometimes we might want to make a logic comparison
> in the conditional statement instead
> that is independent of the data.

Reviewer: ```suggestion
A logic comparison is a type of comparison
that is based on logical rules and conditions,
rather than on the actual data values themselves.
```

> 1. If we are using a ``selection_point``, it would be natural to want to return information about the chosen data point, and then process that information using Python.  This is not currently possible (and as of December 2021 it does not seem likely to become possible any time soon), so any data processing will have to be handled using tools such as ``transform_calculate``, etc. You can follow the progress on this in the following issue: https://github.com/altair-viz/altair/issues/1153.

Reviewer: ```suggestion
1. If we are using a ``selection_point``, it would be natural to want to return information about the chosen data point, and then process that information using Python. This is not currently possible, so any data processing will have to be handled using tools such as ``transform_calculate``, etc. You can follow the progress on this in the following issue: https://github.com/altair-viz/altair/issues/1153.
```
